### PR TITLE
Upgrade to terraform 0.12.x

### PR DESCRIPTION
**Pull request by confusdcodr**

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -1,16 +1,18 @@
...
+}
 
 locals {
   iam_role_name   = "flow-log-${var.vpc_id}"
-  log_group_name  = "${var.log_group_name == "" ? "/aws/vpc/flow-log/${var.vpc_id}" : var.log_group_name}"
-  iam_role_arn    = "${var.iam_role_arn == "" ? join("", aws_iam_role.this.*.arn) : var.iam_role_arn}"
-  create_iam_role = "${var.log_destination_type == "cloud-watch-logs" && var.iam_role_arn == ""}"
+  log_group_name  = var.log_group_name == "" ? "/aws/vpc/flow-log/${var.vpc_id}" : var.log_group_name
+  iam_role_arn    = var.iam_role_arn == "" ? join("", aws_iam_role.this.*.arn) : var.iam_role_arn
+  create_iam_role = var.log_destination_type == "cloud-watch-logs" && var.iam_role_arn == ""
```
Comment (lorengordon) [inline main.tf]: Switch to `null` for all the empty string defaults/tests?

File: tests/baseline_cloudwatch_logs/main.tf [hunk 2]
```diff
@@ -0,0 +1,30 @@
+provider aws {
+  region = "us-east-1"
+}
+
+resource "random_id" "name" {
```
Comment (lorengordon) [inline tests/baseline_cloudwatch_logs/main.tf]: Not seeing this in use?

Comment (confusdcodr) [reply tests/baseline_cloudwatch_logs/main.tf]: Fixed

File: main.tf (hunk 1 above)
Comment (confusdcodr) [reply main.tf]: Fixed

File: variables.tf [hunk 3]
```diff
@@ -4,37 +4,38 @@ variable "create_vpc_flow_log" {
 }
 
 variable "log_destination_type" {
-  description = "Controls whether to create the VPC Flow Log with a CloudWatch Logs or S3 bucket destination"
-  type        = "string"
+  description = "Controls whether to create the VPC Flow Log with a `cloud-watch-logs` or `s3` bucket destination"
+  type        = string
   default     = ""
```
Comment (lorengordon) [inline variables.tf]: Ought to be able to `null` all these empty strings also...

Comment (confusdcodr) [reply variables.tf]: Not able to null the vpc_id without changing the locals given that one `iam_role_name` uses the variable as part of its naming scheme - "flow-log-${var.vpc_id}"

File: main.tf [hunk 4]
```diff
@@ -1,16 +1,18 @@
-provider "aws" {}
+provider "aws" {
+}
 
 locals {
   iam_role_name   = "flow-log-${var.vpc_id}"
-  log_group_name  = "${var.log_group_name == "" ? "/aws/vpc/flow-log/${var.vpc_id}" : var.log_group_name}"
-  iam_role_arn    = "${var.iam_role_arn == "" ? join("", aws_iam_role.this.*.arn) : var.iam_role_arn}"
-  create_iam_role = "${var.log_destination_type == "cloud-watch-logs" && var.iam_role_arn == ""}"
+  log_group_name  = var.log_group_name == null ? "/aws/vpc/flow-log/${var.vpc_id}" : var.log_group_name
```
Comment (lorengordon) [inline main.tf]: Does this should work with a `null` default for var.vpc_id?

```suggestion
  log_group_name  = var.log_group_name == null ? "/aws/vpc/flow-log/${tostring(var.vpc_id)}" : var.log_group_name
```

Comment (lorengordon) [reply main.tf]: no it does not :(

**Pull request merged**